### ISSUE TITLE
External Sorters improvement outline

### ISSUE THREAD

**Issue by h-mayorquin** [SpikeInterface/spikeinterface#1384]
We have been changing recently the way we handle and test sorters on the repo. Terminology: internal sorters are the ones build with sorting components and external sorters are basically the rest as they require external machinery. 

First, @alejoe91 created an in-code separation between internal and external sorters in . #1354
And then we are testing the external sorters with their docker images (right now only for singularity here) CI #1373

This issues it to discuss and then oultine improvements and to-do for the sorters

Here are some random issues and we can outline a to-do later.

* Discuss a policy for which version of spikeinterface the docker images should pull, see [discussion link](https://github.com/SpikeInterface/spikeinterface/pull/1373#issuecomment-1450260167).
* Right now it seems that the Tridesclous is image does not have the pakcage installed, see the latest ran of the action that was created in #1373. 
* Right now we have a dispatch for testing the sorters with singularity #1373. Maybe that should be a weekly cron job?
* Also I feel that the tests at #1373 should run for the whole matrix of OS. This is one of the main purposes of the sorters so it would be really nice to test them for all OS. 
* There should be tests that use docker instead of singularity. There is an [action](https://github.com/marketplace/actions/setup-docker) for that. I can take care of that after I finish the following sorting component round. Should be quick.
* Finally, the tests right now only test that no error is produced when run but there are not expected values. Do some of the sorters have deterministic algorithms (accepting seeds for example) where we can detect their expected values and actually test for that. It would be nice to avoid silent errors. If this is not an option we could create a summary with plots to at least have a visual confirmation that everything is running all right. This can be published in github summary the way that I published the profile timing of the tests.

Feel free to add anything.

**Comment by h-mayorquin** [comment]
Some of this was done here:
https://github.com/SpikeInterface/spikeinterface/pull/1764

Need to come back and check at some point if this issue still makes sense.

**Comment by zm711** [comment]
Coming back to check to see if this still makes sense :)

**Comment by h-mayorquin** [comment]
Yeah, let's close it, hopefully we can come back to it if we get the grant to improve this and add better testing.
